test/integration/src/py/conftest.py

```diff
@@ -79,23 +127,22 @@ def sifchain_fees():
 def source_ethereum_address(is_ropsten_testnet, smart_contracts_dir):
     """account with some starting eth that can be transferred out"""
     addr = test_utilities.get_optional_env_var("ETHEREUM_ADDRESS", "")
+    if addr:
+        logging.debug("using ETHEREUM_ADDRESS provided for source_ethereum_address")
+        return addr
     if is_ropsten_testnet:
+        # Ropsten requires ETHEREUM_ADDRESS
```

Comment:
Seems like ropsten won't return an address. Is this intentional?

Reply:
You have to specify the environment variable if you're using ropsten; this is just an assert that causes the tests to halt if you haven't set it.  It would have returned before you hit the assert if you have it set.  I updated the comment to hopefully make this more clear.